{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<table style=\"border:2px solid white;\" cellspacing=\"0\" cellpadding=\"0\" border-collapse: collapse; border-spacing: 0;>\n",
    "  <tr> \n",
    "    <th style=\"background-color:white\"> <img src=\"../media/ccal-logo-D3.png\" width=225 height=225></th>\n",
    "    <th style=\"background-color:white\"> <img src=\"../media/logoMoores.jpg\" width=175 height=175></th>\n",
    "    <th style=\"background-color:white\"> <img src=\"../media/GP.png\" width=200 height=200></th>\n",
    "    <th style=\"background-color:white\"> <img src=\"../media/UCSD_School_of_Medicine_logo.png\" width=175 height=175></th> \n",
    "    <th style=\"background-color:white\"> <img src=\"../media/Broad.png\" width=130 height=130></th> \n",
    "  </tr>\n",
    "</table>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Part 3 of building a PlatiRes Map**   \n",
    "Daniela Nachmanson *Fall 2017*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using the Chapter 3 of the **Onco-*GPS* notebooks** in order to do NMF with a platinum phenotype derived gene list.\n",
    "The genelist used in this NMF is derived from bulkRNA seq experiment done in Professor Harismendy's as well as an experiment performed in Marchion et al experiment."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<hr style=\"border: none; border-bottom: 3px solid #88BBEE;\">\n",
    "# **Onco-*GPS* Methodology**\n",
    "## **Chapter 3. Annotating the Transcriptional Components**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Authors:** William Kim$^{1}$, Huwate (Kwat) Yeerna$^{2}$, Taylor Cavazos$^{2}$, Kate Medetgul-Ernar$^{2}$, Clarence Mah$^{3}$, Stephanie Ting$^{2}$, Jason Park$^{2}$, Jill P. Mesirov$^{2, 3}$ and Pablo Tamayo$^{2,3}$.\n",
    "\n",
    "**Date:** April 17, 2017\n",
    "\n",
    "1. Eli and Edythe Broad Institute      \n",
    "2. UCSD Moores Cancer Center\n",
    "3. UCSD School of Medicine \n",
    "\n",
    "**Article:** [*Kim et al.* Decomposing Oncogenic Transcriptional Signatures to Generate Maps of Divergent Cellular States](https://drive.google.com/file/d/0B0MQqMWLrsA4b2RUTTAzNjFmVkk/view?usp=sharing)\n",
    "\n",
    "**Analysis overview:** In this chapter we perform a detailed analysis of the platinum-resistance transcriptional components produced by the NMF decomposition in chapter 2 in order to assign a biological interpretation to each component. \n",
    "\n",
    "<img src=\"../media/method_chap3.png\" width=2144 height=1041>\n",
    "\n",
    "The analysis consists of the following steps:\n",
    "* Define a target profile for each component in the CCLE Reference Dataset using the amplitudes of the $H$ matrix. This matrix represents the magnitude of each NMF component per sample. \n",
    "* Using the Information Coefficient (IC) ([*Kim, J.W., Botvinnik 2016*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4868596/)) to estimate the degree of association of each component target profile and several types of genomic features.\n",
    "\n",
    "The genomic features include the following:\n",
    "\n",
    "1. **Mutations and Copy Number Alterations (CNA).** CCLE mutation and copy number datasets (www.broadinstitute.org/ccle, [*Barretina et al. 2012*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3320027/)).\n",
    "2.\t **Gene expression.** CCLE RNA Seq dataset (http://www.broadinstitute.org/ccle, [*Barretina et al. 2012*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3320027/)).\n",
    "3.\t  **Pathway expression** (single sample GSEA of MSigDB gene sets) MSigDB v5.1 sub-collections c2, c5, c6 and h www.msigdb.org, (Liberzon et al. 2011; [*Liberzon et al. 2016. Cell Systems, 1(6), pp.417–425.*](https://www.ncbi.nlm.nih.gov/pubmed/26771021). and a few additional gene sets (see supplementary information in the article).\n",
    "4.\t**Transcription factors and master regulators expression** (single sample GSEA of gene sets) MSigDB v5.1, ([*Liberzon et al. 2011*](https://www.ncbi.nlm.nih.gov/pubmed/21546393)) http://www.msigdb.org, sub-collection c3 and 1,598 IPA gene sets, http://www.ingenuity.com.\n",
    "5.\t **Protein expression.** CCLE Reverse Phased Protein Array (RPPA) dataset (http://www.broadinstitute.org/ccle, [*Barretina et al. 2012*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3320027/)).\n",
    "6.\t **Drug sensitivity.** CCLE dataset (http://www.broadinstitute.org/ccle, [*Barretina et al. 2012*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3320027/))\n",
    "7.\t**Gene dependency.** RNAi Achilles dataset, http://www.broadinstitute.org/achilles, ([*Cowley et al. 2014*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4432652/)).\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<hr style=\"border: none; border-bottom: 3px solid #88BBEE;\">\n",
    "### 1. Set up notebook and import Computational Cancer Analysis Library ([CCAL](https://github.com/KwatME/ccal))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from environment import *\n",
    "\n",
    "%matplotlib inline\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Read the annotation datasets table and $H$  matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The table describing the datasets that will be used in the annotation analysis (annotation.data_table.txt) is included in the directory \"../data.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data Name</th>\n",
       "      <th>Data Type</th>\n",
       "      <th>Emphasis</th>\n",
       "      <th>Filepath</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>drug_sensitivity</td>\n",
       "      <td>continuous</td>\n",
       "      <td>low</td>\n",
       "      <td>../data/ccle_drug_sensitivity.gct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gene_expression</td>\n",
       "      <td>continuous</td>\n",
       "      <td>high</td>\n",
       "      <td>../data/ccle_gene_expression.gct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gene_dependency</td>\n",
       "      <td>continuous</td>\n",
       "      <td>high</td>\n",
       "      <td>../data/ccle_gene_dependency.gct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mutation</td>\n",
       "      <td>binary</td>\n",
       "      <td>high</td>\n",
       "      <td>../data/ccle_mut_CNA.gct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pathway_expression</td>\n",
       "      <td>continuous</td>\n",
       "      <td>high</td>\n",
       "      <td>../data/ccle_pathway_expression_all.gct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>protein_expression</td>\n",
       "      <td>continuous</td>\n",
       "      <td>high</td>\n",
       "      <td>../data/ccle_protein_expression.gct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>regulator</td>\n",
       "      <td>continuous</td>\n",
       "      <td>high</td>\n",
       "      <td>../data/ccle_regulator.gct</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>tissue</td>\n",
       "      <td>binary</td>\n",
       "      <td>high</td>\n",
       "      <td>../data/ccle_tissue.gct</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Data Name   Data Type Emphasis  \\\n",
       "0    drug_sensitivity  continuous      low   \n",
       "1     gene_expression  continuous     high   \n",
       "2     gene_dependency  continuous     high   \n",
       "3            mutation      binary     high   \n",
       "4  pathway_expression  continuous     high   \n",
       "5  protein_expression  continuous     high   \n",
       "6           regulator  continuous     high   \n",
       "7              tissue      binary     high   \n",
       "\n",
       "                                  Filepath  \n",
       "0        ../data/ccle_drug_sensitivity.gct  \n",
       "1         ../data/ccle_gene_expression.gct  \n",
       "2         ../data/ccle_gene_dependency.gct  \n",
       "3                 ../data/ccle_mut_CNA.gct  \n",
       "4  ../data/ccle_pathway_expression_all.gct  \n",
       "5      ../data/ccle_protein_expression.gct  \n",
       "6               ../data/ccle_regulator.gct  \n",
       "7                  ../data/ccle_tissue.gct  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('../data/annotation.data_table.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This function below will read that table and the datasets defined in the \"Filepath\" column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making data bundle for drug_sensitivity ...\n",
      "\tLoaded ../data/ccle_drug_sensitivity.gct.\n",
      "Making data bundle for gene_expression ...\n",
      "\tLoaded ../data/ccle_gene_expression.gct.\n",
      "Making data bundle for gene_dependency ...\n",
      "\tLoaded ../data/ccle_gene_dependency.gct.\n",
      "Making data bundle for mutation ...\n",
      "\tLoaded ../data/ccle_mut_CNA.gct.\n",
      "Making data bundle for pathway_expression ...\n",
      "\tLoaded ../data/ccle_pathway_expression_all.gct.\n",
      "Making data bundle for protein_expression ...\n",
      "\tLoaded ../data/ccle_protein_expression.gct.\n",
      "Making data bundle for regulator ...\n",
      "\tLoaded ../data/ccle_regulator.gct.\n",
      "Making data bundle for tissue ...\n",
      "\tLoaded ../data/ccle_tissue.gct.\n"
     ]
    }
   ],
   "source": [
    "data_table = ccal.load_data_table('../data/annotation.data_table.txt')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " Read the $H$ matrix produced in notebook 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "h_matrix = ccal.read_gct('../output/nmf_cc/nmf/nmf_k7_h.gct')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A101D_SKIN</th>\n",
       "      <th>A172_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>A204_SOFT_TISSUE</th>\n",
       "      <th>A2058_SKIN</th>\n",
       "      <th>A2780_OVARY</th>\n",
       "      <th>A375_SKIN</th>\n",
       "      <th>A498_KIDNEY</th>\n",
       "      <th>A549_LUNG</th>\n",
       "      <th>A673_BONE</th>\n",
       "      <th>A704_KIDNEY</th>\n",
       "      <th>...</th>\n",
       "      <th>WM88_SKIN</th>\n",
       "      <th>WM983B_SKIN</th>\n",
       "      <th>YAPC_PANCREAS</th>\n",
       "      <th>YD10B_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YD38_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YD8_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YH13_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>YKG1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>ZR751_BREAST</th>\n",
       "      <th>ZR7530_BREAST</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C1</th>\n",
       "      <td>232.067182</td>\n",
       "      <td>1478.095344</td>\n",
       "      <td>610.102215</td>\n",
       "      <td>304.444904</td>\n",
       "      <td>1.852530e+03</td>\n",
       "      <td>331.823433</td>\n",
       "      <td>101.448277</td>\n",
       "      <td>357.015088</td>\n",
       "      <td>3100.348060</td>\n",
       "      <td>6.750842</td>\n",
       "      <td>...</td>\n",
       "      <td>370.498254</td>\n",
       "      <td>555.332236</td>\n",
       "      <td>337.854484</td>\n",
       "      <td>462.251312</td>\n",
       "      <td>289.457730</td>\n",
       "      <td>177.374047</td>\n",
       "      <td>932.345000</td>\n",
       "      <td>1634.421092</td>\n",
       "      <td>600.984288</td>\n",
       "      <td>5.163962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C2</th>\n",
       "      <td>0.321277</td>\n",
       "      <td>979.679441</td>\n",
       "      <td>3424.984387</td>\n",
       "      <td>534.758511</td>\n",
       "      <td>4.272116e+03</td>\n",
       "      <td>705.565301</td>\n",
       "      <td>1387.366937</td>\n",
       "      <td>1331.712449</td>\n",
       "      <td>2519.452662</td>\n",
       "      <td>1746.373567</td>\n",
       "      <td>...</td>\n",
       "      <td>816.140078</td>\n",
       "      <td>219.001867</td>\n",
       "      <td>739.033621</td>\n",
       "      <td>471.127722</td>\n",
       "      <td>0.134198</td>\n",
       "      <td>840.432109</td>\n",
       "      <td>26.137021</td>\n",
       "      <td>1081.411271</td>\n",
       "      <td>4329.997888</td>\n",
       "      <td>4737.871000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3</th>\n",
       "      <td>644.842920</td>\n",
       "      <td>1955.798697</td>\n",
       "      <td>2324.925571</td>\n",
       "      <td>1360.520392</td>\n",
       "      <td>2.931981e+03</td>\n",
       "      <td>1746.971910</td>\n",
       "      <td>1760.101873</td>\n",
       "      <td>3468.060422</td>\n",
       "      <td>2263.639368</td>\n",
       "      <td>2289.968057</td>\n",
       "      <td>...</td>\n",
       "      <td>50.866697</td>\n",
       "      <td>916.339951</td>\n",
       "      <td>1538.860277</td>\n",
       "      <td>2069.845640</td>\n",
       "      <td>2524.487797</td>\n",
       "      <td>1406.969205</td>\n",
       "      <td>1497.428196</td>\n",
       "      <td>799.441860</td>\n",
       "      <td>242.309702</td>\n",
       "      <td>0.027625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C4</th>\n",
       "      <td>186.648538</td>\n",
       "      <td>19.566442</td>\n",
       "      <td>38.619616</td>\n",
       "      <td>372.349661</td>\n",
       "      <td>6.124098e+00</td>\n",
       "      <td>492.214407</td>\n",
       "      <td>953.565669</td>\n",
       "      <td>920.808159</td>\n",
       "      <td>139.932584</td>\n",
       "      <td>2211.291331</td>\n",
       "      <td>...</td>\n",
       "      <td>225.358655</td>\n",
       "      <td>161.965636</td>\n",
       "      <td>2335.152300</td>\n",
       "      <td>280.133274</td>\n",
       "      <td>794.193296</td>\n",
       "      <td>95.562497</td>\n",
       "      <td>362.996504</td>\n",
       "      <td>229.054029</td>\n",
       "      <td>2742.021757</td>\n",
       "      <td>3134.720759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C5</th>\n",
       "      <td>1379.580409</td>\n",
       "      <td>4059.425214</td>\n",
       "      <td>1739.160372</td>\n",
       "      <td>879.096088</td>\n",
       "      <td>7.569543e+02</td>\n",
       "      <td>913.214162</td>\n",
       "      <td>3375.133431</td>\n",
       "      <td>1625.953739</td>\n",
       "      <td>1389.711154</td>\n",
       "      <td>2413.700490</td>\n",
       "      <td>...</td>\n",
       "      <td>936.252055</td>\n",
       "      <td>958.728154</td>\n",
       "      <td>677.965254</td>\n",
       "      <td>1262.685186</td>\n",
       "      <td>890.344381</td>\n",
       "      <td>4404.869902</td>\n",
       "      <td>5426.175363</td>\n",
       "      <td>3498.336481</td>\n",
       "      <td>0.770284</td>\n",
       "      <td>303.328827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C6</th>\n",
       "      <td>8190.890283</td>\n",
       "      <td>1008.767726</td>\n",
       "      <td>1851.351694</td>\n",
       "      <td>7205.088176</td>\n",
       "      <td>1.724504e+02</td>\n",
       "      <td>6362.994378</td>\n",
       "      <td>615.308777</td>\n",
       "      <td>715.852830</td>\n",
       "      <td>917.742018</td>\n",
       "      <td>270.281574</td>\n",
       "      <td>...</td>\n",
       "      <td>8564.957030</td>\n",
       "      <td>7943.831754</td>\n",
       "      <td>330.214109</td>\n",
       "      <td>811.646059</td>\n",
       "      <td>846.579874</td>\n",
       "      <td>171.418896</td>\n",
       "      <td>1140.103031</td>\n",
       "      <td>3224.551292</td>\n",
       "      <td>650.885735</td>\n",
       "      <td>385.246200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C7</th>\n",
       "      <td>591.403983</td>\n",
       "      <td>350.068929</td>\n",
       "      <td>61.445347</td>\n",
       "      <td>369.370027</td>\n",
       "      <td>3.604511e-08</td>\n",
       "      <td>279.630035</td>\n",
       "      <td>1331.956109</td>\n",
       "      <td>1091.132989</td>\n",
       "      <td>0.056211</td>\n",
       "      <td>505.515183</td>\n",
       "      <td>...</td>\n",
       "      <td>486.540362</td>\n",
       "      <td>492.042817</td>\n",
       "      <td>3573.093226</td>\n",
       "      <td>4227.383185</td>\n",
       "      <td>4151.553681</td>\n",
       "      <td>2305.468819</td>\n",
       "      <td>312.526337</td>\n",
       "      <td>6.527126</td>\n",
       "      <td>1476.087208</td>\n",
       "      <td>1333.166071</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7 rows × 750 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     A101D_SKIN  A172_CENTRAL_NERVOUS_SYSTEM  A204_SOFT_TISSUE   A2058_SKIN  \\\n",
       "C1   232.067182                  1478.095344        610.102215   304.444904   \n",
       "C2     0.321277                   979.679441       3424.984387   534.758511   \n",
       "C3   644.842920                  1955.798697       2324.925571  1360.520392   \n",
       "C4   186.648538                    19.566442         38.619616   372.349661   \n",
       "C5  1379.580409                  4059.425214       1739.160372   879.096088   \n",
       "C6  8190.890283                  1008.767726       1851.351694  7205.088176   \n",
       "C7   591.403983                   350.068929         61.445347   369.370027   \n",
       "\n",
       "     A2780_OVARY    A375_SKIN  A498_KIDNEY    A549_LUNG    A673_BONE  \\\n",
       "C1  1.852530e+03   331.823433   101.448277   357.015088  3100.348060   \n",
       "C2  4.272116e+03   705.565301  1387.366937  1331.712449  2519.452662   \n",
       "C3  2.931981e+03  1746.971910  1760.101873  3468.060422  2263.639368   \n",
       "C4  6.124098e+00   492.214407   953.565669   920.808159   139.932584   \n",
       "C5  7.569543e+02   913.214162  3375.133431  1625.953739  1389.711154   \n",
       "C6  1.724504e+02  6362.994378   615.308777   715.852830   917.742018   \n",
       "C7  3.604511e-08   279.630035  1331.956109  1091.132989     0.056211   \n",
       "\n",
       "    A704_KIDNEY      ...          WM88_SKIN  WM983B_SKIN  YAPC_PANCREAS  \\\n",
       "C1     6.750842      ...         370.498254   555.332236     337.854484   \n",
       "C2  1746.373567      ...         816.140078   219.001867     739.033621   \n",
       "C3  2289.968057      ...          50.866697   916.339951    1538.860277   \n",
       "C4  2211.291331      ...         225.358655   161.965636    2335.152300   \n",
       "C5  2413.700490      ...         936.252055   958.728154     677.965254   \n",
       "C6   270.281574      ...        8564.957030  7943.831754     330.214109   \n",
       "C7   505.515183      ...         486.540362   492.042817    3573.093226   \n",
       "\n",
       "    YD10B_UPPER_AERODIGESTIVE_TRACT  YD38_UPPER_AERODIGESTIVE_TRACT  \\\n",
       "C1                       462.251312                      289.457730   \n",
       "C2                       471.127722                        0.134198   \n",
       "C3                      2069.845640                     2524.487797   \n",
       "C4                       280.133274                      794.193296   \n",
       "C5                      1262.685186                      890.344381   \n",
       "C6                       811.646059                      846.579874   \n",
       "C7                      4227.383185                     4151.553681   \n",
       "\n",
       "    YD8_UPPER_AERODIGESTIVE_TRACT  YH13_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "C1                     177.374047                   932.345000   \n",
       "C2                     840.432109                    26.137021   \n",
       "C3                    1406.969205                  1497.428196   \n",
       "C4                      95.562497                   362.996504   \n",
       "C5                    4404.869902                  5426.175363   \n",
       "C6                     171.418896                  1140.103031   \n",
       "C7                    2305.468819                   312.526337   \n",
       "\n",
       "    YKG1_CENTRAL_NERVOUS_SYSTEM  ZR751_BREAST  ZR7530_BREAST  \n",
       "C1                  1634.421092    600.984288       5.163962  \n",
       "C2                  1081.411271   4329.997888    4737.871000  \n",
       "C3                   799.441860    242.309702       0.027625  \n",
       "C4                   229.054029   2742.021757    3134.720759  \n",
       "C5                  3498.336481      0.770284     303.328827  \n",
       "C6                  3224.551292    650.885735     385.246200  \n",
       "C7                     6.527126   1476.087208    1333.166071  \n",
       "\n",
       "[7 rows x 750 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h_matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3. Find the top genomic features that match each component profile\n",
    "The annotation consists of running the association analysis for each component against all the genomic datasets. Because this a double iteration over componets and feature datasets it will take hours to complete. As the program runs it will display the specific target vs. features comparison being made."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C1 vs Regulator ...\n",
      "Created directory /Users/DanielaNachmanson/Desktop/OncoGPS_Analysis_paper/dana/results/component_annotation.\n",
      "C1 vs Gene Expression ...\n",
      "C1 vs Mutation ...\n",
      "C1 vs Protein Expression ...\n",
      "C1 vs Pathway Expression ...\n",
      "C1 vs Tissue ...\n",
      "C1 vs Gene Dependency ...\n",
      "C1 vs Drug Sensitivity ...\n",
      "C2 vs Regulator ...\n",
      "C2 vs Gene Expression ...\n",
      "C2 vs Mutation ...\n",
      "C2 vs Protein Expression ...\n",
      "C2 vs Pathway Expression ...\n",
      "C2 vs Tissue ...\n",
      "C2 vs Gene Dependency ...\n",
      "C2 vs Drug Sensitivity ...\n",
      "C3 vs Regulator ...\n",
      "C3 vs Gene Expression ...\n",
      "C3 vs Mutation ...\n"
     ]
    }
   ],
   "source": [
    "ccal.association.make_association_panels(target=h_matrix, data_bundle = data_table, dropna='all', target_ascending=False,\n",
    "                            target_prefix='', data_prefix='', target_type='continuous', n_jobs=1, n_features=20, n_samplings=30, \n",
    "                            n_permutations=50, random_seed=12345,directory_path='../output/component_annotation')"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
